modules/advanced_csv_to_commands.py

```diff
@@ -0,0 +1,139 @@
+"""
+An command generator with advanced parameters and greater flexibility
+"""
+
+import pathlib
+
+import dronekit
+
+
+VALID_FRAMES = {
+    "global": dronekit.mavutil.mavlink.MAV_FRAME_GLOBAL,
+    "global_relative_alt": dronekit.mavutil.mavlink.MAV_FRAME_GLOBAL_RELATIVE_ALT,
+}
+
+VALID_COMMANDS = {
+    "land": dronekit.mavutil.mavlink.MAV_CMD_NAV_LAND,
+    "return_to_launch": dronekit.mavutil.mavlink.MAV_CMD_NAV_RETURN_TO_LAUNCH,
+    "takeoff": dronekit.mavutil.mavlink.MAV_CMD_NAV_TAKEOFF,
+    "waypoint": dronekit.mavutil.mavlink.MAV_CMD_NAV_WAYPOINT,
+    "waypoint_spline": dronekit.mavutil.mavlink.MAV_CMD_NAV_SPLINE_WAYPOINT,
+    "loiter_timed": dronekit.mavutil.mavlink.MAV_CMD_NAV_LOITER_TIME,
+    "loiter_unlimited": dronekit.mavutil.mavlink.MAV_CMD_NAV_LOITER_UNLIM,
+    "do_jump": dronekit.mavutil.mavlink.MAV_CMD_DO_JUMP,
+}
+
+
+def generate_command_advanced(
+    frame: str,
+    command_type: str,
+    param1: float,  # I'm anxious about this because of do_jump which requires ints
```

Comment:
This function can dispatch to more specialized functions to construct the various commands.